### ISSUE TITLE
Include first 6 lines of the source code

### ISSUE THREAD

**Issue by nikgraf** [nikgraf/react-hooks#9]

**Comment by stereobooster** [comment]
Separate page per hook? Then it would be possible to include whole text. For example, https://usehooks.com/useDarkMode/

**Comment by nikgraf** [comment]
@stereobooster yeah, either a separate page or a section that's collapsed and can be expanded. The issue though is that some hooks might have multiple hundred lines of code. Also is it done on build time or do we fetch it on demand? An iFrame of that Github file could also do it.

**Comment by stereobooster** [comment]
> The issue though is that some hooks might have multiple hundred lines of code.

Let's limit scope. We will care only about hooks :

- implemented in JS as first step, in TS as second step, everything else later
- self contained hooks e.g. we allow imports from packages, but don't allow imports from different files
- one hook per file
- named export, so we can identify hook by `use...`
- no hooks generators, like `generateUseState` etc.
- everything what doesn't fit constraints will be disqualified e.g. we will have link, but no code preview

### How can we get source, and how we will keep it in sync?

...to be continued

**Comment by nikgraf** [comment]
sounds good to me!

Regarding how we can get the source. In every Hook object we could link to the raw file and during build time fetch it. We could trigger a build every hour or two to keep it up to date.

What do you think?

**Comment by stereobooster** [comment]
yes good idea. I don't think we need to trigger every two hours more like once a day - once a week (but this can be configured).

### Metadata and information that can be extracted

Github link - stars, fork, open issues, last updated etc
npm package name - latest published version, downloads, date of publishing
link to the raw file - source code, external dependencies, core hooks dependencies, LOC

### Script to extract data

- read hooks.json
- convert it to stream https://highlandjs.org/
- filter items with metada
- flatmap over stream with async function (`fetch`)
- log all failures, like 404 or 500
- write output

**Comment by stereobooster** [comment]
We have link to raw files (for some hooks) and we already fetch source to detect subhooks on each CI build, just in case anybody wants to pick up this task

**Comment by aarohmankad** [comment]
For files that have the raw source (`sourceUrl`), I can't seem to just `fetch` that url.

**Comment by dmitryrn** [comment]
I would like to improve something here. I already made first-6-lines-source-preview. Is this relevant to make PR here?
![image](https://user-images.githubusercontent.com/25539852/66143695-4996de00-e610-11e9-9a3d-0c99112c0401.png)

**Comment by dmitryrn** [comment]
@stereobooster I guess I’ll pick it up

**Comment by nikgraf** [comment]
@dmitryrn cool experiment. maybe rather showing the first 6 lines of code it would be good to have some insights on how many lines there are and possibly expand it to see the source?

What do you think? (keep in mind this are rough thoughts and need experimentation)
